### PR TITLE
[#43670] add OpenProject to the files action menu in NC

### PR DESCRIPTION
**Pull request by kiranparajuli589**
### Description
Added OpenProject button inside the file actions menu with opens the OpenProject Sidebar Tab.

### Related
https://community.openproject.org/work_packages/43670
Signed-off-by: Kiran Parajuli <kiranparajuli589@gmail.com>

### CODE REVIEW

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -68,6 +70,12 @@ public function register(IRegistrationContext $context): void {
 
 	public function boot(IBootContext $context): void {
 		$context->injectFn(Closure::fromCallable([$this, 'registerNavigation']));
+		/** @var IEventDispatcher $dispatcher */
+		$dispatcher = $context->getAppContainer()->get(IEventDispatcher::class);
+		// TODO this creates some deprecation warnings!
```
Comment (individual-it) [inline lib/AppInfo/Application.php:75]: a TODO in the code will not be worked on and will be forgotten.
Ether fix this now or create a work-package

File: src/fileActions.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	permissions: OC.PERMISSION_READ,
+	iconClass: 'icon-openproject',
+	actionHandler: (filename, context) => {
+		const fileList = context.fileList
+		if (!fileList._detailsView) {
+			return
+		}
+		// use the sidebar-tab id for the navigation
+		fileList.showDetailsView(filename, 'open-project')
+	}
```
Comment (individual-it) [inline src/fileActions.js:16]:
```suggestion
	},
```
to make the linter happy

**Review by individual-it — CHANGES_REQUESTED** [review]
works fine, just some small comments

File: lib/AppInfo/Application.php (hunk 1 above)
Comment (kiranparajuli589) [reply lib/AppInfo/Application.php]: sorry, the comment was not relevant. removed it.

**Review by individual-it — APPROVED** [review]

**Pull request merged**